src/metabase/api/action.clj

```diff
@@ -115,24 +117,23 @@
       (last (action/select-actions nil :type type)))))
 
 (api/defendpoint PUT "/:id"
-  [id :as
-   {{:keys [archived database_id dataset_query description error_handle kind model_id name parameter_mappings parameters
-            response_handle template type visualization_settings] :as action} :body}]
-  {id                     pos-int?
-   archived               [:maybe boolean?]
-   database_id            [:maybe pos-int?]
-   dataset_query          [:maybe map?]
-   description            [:maybe :string]
-   error_handle           [:maybe json-query-schema]
-   kind                   [:maybe implicit-action-kind]
-   model_id               [:maybe pos-int?]
-   name                   [:maybe :string]
-   parameter_mappings     [:maybe map?]
-   parameters             [:maybe [:sequential map?]]
-   response_handle        [:maybe json-query-schema]
-   template               [:maybe http-action-template]
-   type                   [:maybe supported-action-type]
-   visualization_settings [:maybe map?]}
+  [id :as {action :body}]
+  {id     ms/Id
+   action [:map
+           [:archived               {:optional true} [:maybe :boolean]]
+           [:database_id            {:optional true} [:maybe ms/Id]]
+           [:dataset_query          {:optional true} [:maybe :map]]
+           [:description            {:optional true} [:maybe :string]]
+           [:error_handle           {:optional true} [:maybe json-query-schema]]
+           [:kind                   {:optional true} [:maybe implicit-action-kind]]
+           [:model_id               {:optional true} [:maybe ms/Id]]
+           [:name                   {:optional true} [:maybe :string]]
+           [:parameter_mappings     {:optional true} [:maybe :map]]
+           [:parameters             {:optional true} [:maybe [:sequential :map]]]
+           [:response_handle        {:optional true} [:maybe json-query-schema]]
+           [:template               {:optional true} [:maybe http-action-template]]
+           [:type                   {:optional true} [:maybe supported-action-type]]
+           [:visualization_settings {:optional true} [:maybe :map]]]}
```

Comment:
I want to highlight something that can work pretty well now. It felt artificial to force this destructuring into a single flat level and not use any of the values. But, if this will cause an issue I can revert it.